src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useCallback, useMemo } from 'react';
+import { debounce } from 'lodash';
+
+import { peopleFromServer } from '../../data/people';
+
+import { SearchBar } from '../SearchBar';
+import { DropdownList } from '../DropdownList';
+import { Person } from '../../types/Person';
+
+function filterPeopleByQuery(peopleToFilter: Person[], query: string) {
+  const normalisedQuery = query.toLowerCase();
+
+  return peopleToFilter
+    .filter(({ name }) => name.toLocaleLowerCase().includes(normalisedQuery));
+}
+
+type Props = {
+  setSelectedPerson: (person: Person) => void,
+  delay: number,
+};
+
+export const Autocomplete: React.FC<Props> = ({ setSelectedPerson, delay }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const applyQuery
+  = useCallback(debounce((value: string) => {
```

Comment:
This `useCallback` looks unnecessary

Reply:
It is needed to prevent extra rerendering (the callback creates every time the query changes)

Reply:
`useCallback` only does one thing - it stores the current function object, so it survives renders, unless the dependencies change.

It is useful in basically two scenarios:
- It is the dependency for another hook (like useMemo, useCallback, useEffect) 
- It is passed as a property to the memorized component

In your case, you just use `applyQuery` in another callback function, so usage of the useCallback has no benefits. Even more - any memorization without clear gain is harmful, as it adds code complexity, degrades performance, and adds a perfect space for hard-to-trace bugs.